internal/download/download.go

```diff
@@ -71,3 +81,97 @@ func DownloadFile(url, outPath, header, token string) error {
 
 	return nil
 }
+
+// DownloadRegistryOpenAPIBundle Returns a file path within the downloaded bundle or error
+func DownloadRegistryOpenAPIBundle(ctx context.Context, namespaceID, reference, outPath string) (string, error) {
+	serverURL := auth.GetServerURL()
+	insecurePublish := false
+	if strings.HasPrefix(serverURL, "http://") {
+		insecurePublish = true
+	}
+	reg := strings.TrimPrefix(serverURL, "http://")
+	reg = strings.TrimPrefix(reg, "https://")
+
+	bundleLoader := loader.NewLoader(loader.OCILoaderOptions{
+		Registry: reg,
+		Access: ocicommon.NewRepositoryAccess(config.GetSpeakeasyAPIKey(), namespaceID, ocicommon.RepositoryAccessOptions{
+			Insecure: insecurePublish,
+		}),
+	})
+
+	bundleResult, err := bundleLoader.LoadOpenAPIBundle(ctx, reference)
+	if err != nil {
+		return "", err
+	}
+
+	buf, err := io.ReadAll(bundleResult.Body)
+	if err != nil {
+		return "", err
+	}
+	defer bundleResult.Body.Close()
```

Comment:
this defer should be moved to after the error check following `bundleLoader.LoadOpenAPIBundle`